cmd/quadlet/main.go

```diff
@@ -49,12 +49,13 @@ var (
 	// Key: Extension
 	// Value: Processing order for resource naming dependencies
 	supportedExtensions = map[string]int{
-		".container": 3,
+		".container": 4,
```

Comment:
It seems we have a cyclic dependency problem between `.volume` and `.build` files. A `.volume` can depend on a `.build` file for it's `Image` key while a `.build` file can depend on `.volume` file for its `Volume` key. I'm not talking about the same files (that's the user's problem), but on different couples of files.

I had a similar issue when I added the support for `.pod` files. The way I solved it was by doing an initial run on `.pod` files (See `generatePodsInfoMap`) and passing it around.

Reply:
Good catch! Yes, this seems like a tricky issue. So I think I need to do two additional runs similar to how you've handled a pod's containers:

1. Check all .volumes for referenced .build files in the Image Key
2. Check all .build files for referenced .volume files in a Volume Key, and here also bail out in case the user added a cyclic dependency

Reply:
We try not to have Quadlet catch user errors. So, I don't think you need to handle any bailouts.
I think a simpler solution would be to calculate the `resourceNames` map in a preliminary run and then do the full conversion. At the end of the day you just need the names, Quadlet does not run anything

Reply:
Yes, I agree that such a change should have its own PR

Reply:
Thanks. I will thus add the missing test cases next.